File: code/powerplant-coding-challenge/powerplant-coding-challenge/WindFactor.cs
namespace powerplant_coding_challenge
{
    /// <summary>
    /// This is the class used to add a wind factor to a productor.
    /// </summary>
    public class WindFactor : IPhysicFactor
    {

        // -------------- Constructors --------------
        /// <summary>
        /// This is the main constructor of the wind factor object.
        /// </summary>
        /// <param name="windValue">This is the wind value.</param>
        public WindFactor (Double windValue)
        {
            this.PhysicValue = windValue;
        }

        // -------------- Methods --------------

        public double ComputePower(double powerInput)
        {
            return powerInput * this.PhysicValue;
        }


        // -------------- Methods --------------

        public Double PhysicValue
        {
            get;
            set;
        } = 0;
    }
}

File: code/python/windfactor.py
#-----------------------------------
# Imports
#-----------------------------------

from iphysicfactor import IPhysicFactor

#-----------------------------------
# Class
#-----------------------------------

class WindFactor (IPhysicFactor) :
    """This is the wind physic factor."""

    # -------------
    # Fields
    # -------------

    __wind_factor__: float = 0.0
    """This is the wind factor to apply to the power plants."""

    # -------------
    # Constructors
    # -------------

    def __init__ (self, wind_factor: float) :
        self.__wind_factor__ = wind_factor

    # -------------
    # Methods
    # -------------

    def compute_power (self, power_input: float) -> float :
        return power_input * self.__wind_factor__

    def get_efficiency (self) -> float :
        return self.__wind_factor__

    # -------------
    # Getters
    # -------------

    def get_wind_factor (self) -> float :
        """This method allows to retrieve the wind factor. Returns a float representing the wind factor."""
        return self.__wind_factor__

